### DISCUSSION TITLE
Mass import files ?

### DISCUSSION THREAD

**Discussion by dev-a** [TandoorRecipes/recipes#1153] [Q&A]
Hi 
I'm thinking about importing many recipes stored on a pc in various formats, but mainly HTML.

I'm thinking of developing a script to get the right information (name, ingredients, recipe, tag, etc..).

I tried to look at the documentation but I did not see if there was a way to import recipes in mass for example via a webservice.

Is this use case possible? 
Sincerely

**Comment by smilerz** [comment]
There are 2 apis that would be helpful.
The first is `/api/recipe-from-source/` which is undocumented - but can take a POST and create a recipe:
```
{
    url: String: url of the original recipe
    data: String: HTML body or ld-json
    mode: 'source' # tells api to create recipe from source text
    auto: 'true' # tells api to create recipe false will return a JSON document of the recipe
}
```

The other is the recipe end point which is documented in full at `/recipes/docs/api/#operation/createRecipe` or `/openapi`

**Comment by dev-a** [comment]
Ok I will look at all of these

Thanks for the quick response
